raiko-guests/powdr/src/main.rs

```diff
@@ -1,11 +1,61 @@
 #![no_std]
 
 extern crate alloc;
-use alloc::{collections::BTreeMap, vec};
+use alloc::{collections::BTreeMap, string::ToString, vec};
+
+use powdr_riscv_runtime::{self, coprocessors::get_data_serde};
+use serde::Deserialize;
+use zeth_lib::{
+    builder::{BlockBuilderStrategy, TaikoStrategy},
+    consts::{ChainSpec, TKO_MAINNET_CHAIN_SPEC},
+    input::Input,
+    taiko::{
+        host::{init_taiko, HostArgs},
+        protocol_instance::{assemble_protocol_instance, EvidenceType},
+        TaikoSystemInfo,
+    },
+    EthereumTxEssence,
+};
 use zeth_primitives::U256;
-use zeth_lib::consts::ChainSpec;
-use powdr_riscv_runtime;
+
+#[derive(Deserialize)]
+struct InputData {
+    host_args: HostArgs,
+    l2_chain_spec: ChainSpec,
+    testnet: String,
+    l2_block_no: u64,
+    graffiti: B256,
+    prover: Address,
+}
 
 #[no_mangle]
 fn main() {
-}
\ No newline at end of file
+    let InputData {
+        host_args,
+        l2_chain_spec,
+        testnet,
+        l2_block_no,
+        graffiti,
+        prover,
+    } = get_data_serde::<InputData>(42);
```

Comment:
Have u figured out what does this number "42" mean? I'm asking cuz I don't really know so not sure if it will break when we actually run with real block.

Reply:
It's a channel on which the input is sent, so we add it here on the receiving and and it must match on the sending end [here](https://github.com/petarvujovic98/zeth/blob/95b9a091758e096e8275444efa0b601acbca5d54/raiko-host/src/prover/proof/powdr.rs#L74). I recommend we use a shared const somewhere in the code base so that this remains synced instead of being manually hardcoded

Reply:
https://en.wikipedia.org/wiki/Phrases_from_The_Hitchhiker%27s_Guide_to_the_Galaxy#The_Answer_to_the_Ultimate_Question_of_Life,_the_Universe,_and_Everything_is_42